sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_container_registry_client.py

```diff
@@ -839,6 +847,62 @@ async def update_tag_properties(
             repository=repository
         )
 
+    @distributed_trace_async
+    async def upload_blob(self, repository: str, data: IO[bytes], **kwargs) -> Tuple[str, int]:
+        """Upload an artifact blob.
+
+        :param str repository: Name of the repository.
+        :param data: The blob to upload. Note: This must be a seekable stream.
+        :type data: IO
+        :returns: The digest and size of the uploaded blob.
+        :rtype: Tuple[str, int]
+        :raises ValueError: If the parameter repository or data is None.
+        """
+        try:
+            start_upload_response_headers = cast(
+                Dict[str, str],
+                await self._client.container_registry_blob.start_upload(
+                    repository, cls=_return_response_headers, **kwargs
+                )
+            )
+            digest, location, blob_size = await self._upload_blob_chunk(
+                start_upload_response_headers['Location'], data, **kwargs
+            )
+            complete_upload_response_headers = cast(
+                Dict[str, str],
+                await self._client.container_registry_blob.complete_upload(
+                    digest=digest,
+                    next_link=location,
+                    cls=_return_response_headers,
+                    **kwargs
+                )
+            )
+        except Exception:
+            if repository is None or data is None:
+                raise ValueError("The parameter repository and data cannot be None.")
+            raise
```

Comment:
I've only seen this kind of custom error message raising in places where a `None` value would really return a strange error (e.g. when the parameter is part of the url path and causes a 404). Otherwise I'm not sure if it's needed.

Reply:
Yeah, I don't see the `None` check is widely do in each function. In my test, the error message is not that clear as I replied [above](https://github.com/Azure/azure-sdk-for-python/pull/29382#discussion_r1145207313)